recipes-bsp/u-boot/files/0011-board-siemens-iot2050-get-and-set-m.2-releated-infor.patch

```diff
@@ -0,0 +1,229 @@
++{
++	int i,j;
++	serdes_config_t serdes_config = {USB30, PCIE1_LANE0};//default
++	int config_table_size= sizeof(config_table)/sizeof(config_table[0]);
++
++    //1.TBD:check eeprom or flash
++    //2.configure the m2
++	int *config = Get_M2_Config();
++	printf("M2-KEY PIN:");
++	for(i=0; i < M2_CONFIG_LEN;i++)
++	{
++		printf("%d",*(config+i));
++	}
++	printf("\n");
++
++	for(i = 0; i < config_table_size; i++)
++	{
++		if(!memcmp(config, config_table[i].m2config, M2_CONFIG_LEN * sizeof(int)))
++		{
++			// set m2 pinmux
++			for(j = 0; j < PCIE_MUX_CONTROL; j++)
++				Set_PinValue(m2_pinmux_config[j].gpio_name, m2_pinmux_config[j].label, config_table[i].m2_pinmux_value[j]);
++			//get serdes config info
++			serdes_config =  config_table[i].serdes_config;
++			break;
++		}
++	}
++
++	return serdes_config;
++}
++
++
+ static bool board_is_advanced(void)
+ {
+ 	struct iot2050_info *info = IOT2050_INFO_DATA;
+@@ -60,7 +195,16 @@ static bool board_is_sr1(void)
+ 	struct iot2050_info *info = IOT2050_INFO_DATA;
+ 
+ 	return info->magic == IOT2050_INFO_MAGIC &&
+-		!strstr((char *)info->name, "-PG2");
++		!strstr((char *)info->name, "-PG2") &&
++		!strstr((char *)info->name, "-M2");
++}
++
++static bool board_is_m2(void)
++{
++	struct iot2050_info *info = IOT2050_INFO_DATA;
++
++	return info->magic == IOT2050_INFO_MAGIC &&
++		strstr((char *)info->name, "-M2") != NULL;
+ }
+ 
+ static void remove_mmc1_target(void)
+@@ -76,6 +220,28 @@ static void remove_mmc1_target(void)
+ 	free(boot_targets);
+ }
+ 
++static void set_m2_info(void)
++{
++	struct iot2050_info *info = IOT2050_INFO_DATA;
++	serdes_config_t serdes_conf;
++    /*IOT2050 M2 Variant*/
++	if(strcmp(info->name, "IOT2050-ADVANCED-M2") == 0) {
++		//Enable Power of E-key and B-key connector
++		Set_PinValue(m2_enable_power.gpio_name, m2_enable_power.label, 1);
++        serdes_conf = Get_Serdes_Config_Set_M2_Pinmux();
++		if (serdes_conf.lane0_sel == PCIE0_LANE0 && serdes_conf.lane1_sel == PCIE0_LANE1)
++			env_set("name_overlays", "k3-am654-iot2050-bkey-pciex2.dtbo");
++		else if (serdes_conf.lane0_sel == USB30 && serdes_conf.lane1_sel == PCIE0_LANE1)
++			env_set("name_overlays", "k3-am654-iot2050-bkey-usb3-ekey-pcie.dtbo");
++		else if (serdes_conf.lane0_sel == PCIE0_LANE0 && serdes_conf.lane1_sel == PCIE1_LANE0)
++			env_set("name_overlays", "k3-am654-iot2050-bkey-pcie-ekey-pcie.dtbo");
++		else 
++			env_set("name_overlays", "k3-am654-iot2050-bkey-usb3-ekey-pcie.dtbo");
++
++		env_save();
++        printf("serdes0: %d serdes1 %d \n",serdes_conf.lane0_sel,serdes_conf.lane1_sel);
++	}
++}
+ void set_board_info_env(void)
+ {
+ 	struct iot2050_info *info = IOT2050_INFO_DATA;
+@@ -111,6 +277,8 @@ void set_board_info_env(void)
+ 	if (board_is_advanced()) {
+ 		if (board_is_sr1())
+ 			fdtfile = "ti/k3-am6548-iot2050-advanced.dtb";
++		else if (board_is_m2())
++			fdtfile = "ti/k3-am6548-iot2050-advanced-m2.dtb";
+ 		else
+ 			fdtfile = "ti/k3-am6548-iot2050-advanced-pg2.dtb";
```

Comment:
FYI, the `board_is_sr1()` is about to be abandoned soon, in the future the sr1 and sr2 will have separate building, and we will not hold runtime checking for sr1 and sr2.

 See #236